contracts/src/v0.8/ccip/libraries/ERC165CheckerReverting.sol

```diff
@@ -0,0 +1,142 @@
+ * if < 31_000 gas is available, to prevent message delivery issues in CCIP.
+ */
+library ERC165CheckerReverting {
+  // As per the EIP-165 spec, no interface should ever match 0xffffffff
+  bytes4 private constant _INTERFACE_ID_INVALID = 0xffffffff;
+
+  // bytes4(keccak256("NotEnoughGasForSupportsInterfaceCall()"))
+  bytes4 private constant NOT_ENOUGH_GAS_SIG = 0x161c3bf7;
+
+  /**
+   * @dev Returns true if `account` supports the {IERC165} interface.
+   */
+  function _supportsERC165Reverting(
+    address account
+  ) internal view returns (bool) {
+    // Any contract that implements ERC165 must explicitly indicate support of
+    // InterfaceId_ERC165 and explicitly indicate non-support of InterfaceId_Invalid
+    return _supportsERC165InterfaceUncheckedReverting(account, type(IERC165).interfaceId)
+      && !_supportsERC165InterfaceUncheckedReverting(account, _INTERFACE_ID_INVALID);
+  }
+
+  /**
+   * @dev Returns true if `account` supports the interface defined by
+   * `interfaceId`. Support for {IERC165} itself is queried automatically.
+   *
+   * See {IERC165-supportsInterface}.
+   */
+  function _supportsInterfaceReverting(address account, bytes4 interfaceId) internal view returns (bool) {
+    // query support of both ERC165 as per the spec and support of _interfaceId
+    return _supportsERC165Reverting(account) && _supportsERC165InterfaceUncheckedReverting(account, interfaceId);
+  }
+
+  /**
+   * @dev Returns a boolean array where each value corresponds to the
+   * interfaces passed in and whether they're supported or not. This allows
+   * you to batch check interfaces for a contract where your expectation
+   * is that some interfaces may not be supported.
+   *
+   * See {IERC165-supportsInterface}.
+   *
+   * _Available since v3.4._
+   */
+  function _getSupportedInterfacesReverting(
+    address account,
+    bytes4[] memory interfaceIds
+  ) internal view returns (bool[] memory) {
+    // an array of booleans corresponding to interfaceIds and whether they're supported or not
+    bool[] memory interfaceIdsSupported = new bool[](interfaceIds.length);
+
+    // query support of ERC165 itself
+    if (_supportsERC165Reverting(account)) {
+      // query support of each interface in interfaceIds
+      for (uint256 i = 0; i < interfaceIds.length; i++) {
+        interfaceIdsSupported[i] = _supportsERC165InterfaceUncheckedReverting(account, interfaceIds[i]);
+      }
+    }
+
+    return interfaceIdsSupported;
+  }
+
+  /**
+   * @dev Returns true if `account` supports all the interfaces defined in
+   * `interfaceIds`. Support for {IERC165} itself is queried automatically.
+   *
+   * Batch-querying can lead to gas savings by skipping repeated checks for
+   * {IERC165} support.
+   *
+   * See {IERC165-supportsInterface}.
+   */
+  function _supportsAllInterfacesReverting(address account, bytes4[] memory interfaceIds) internal view returns (bool) {
```

Comment:
can probably keep this version minimal and only include things we're actually using